cmd/healthinfo.go

```diff
@@ -19,329 +19,92 @@ package cmd
+	if n <= 0 {
+		return math.Round(f)
 	}
 
+	p := math.Pow10(n)
+	return math.Round(f*p) / p
 }
 
-func getLocalDrives(ctx context.Context, parallel bool, endpointServerPools EndpointServerPools, r *http.Request) madmin.ServerDrivesInfo {
-	var drivesPerfInfo []madmin.DrivePerfInfo
+func getDiskPerfInfo(ctx context.Context, parallel bool) []madmin.DiskPerfInfo {
+	pools := globalEndpoints
+	info := []madmin.DiskPerfInfo{}
 	var wg sync.WaitGroup
-	for _, ep := range endpointServerPools {
-		for _, endpoint := range ep.Endpoints {
-			// Only proceed for local endpoints
-			if endpoint.IsLocal {
-				if _, err := os.Stat(endpoint.Path); err != nil {
-					// Since this drive is not available, add relevant details and proceed
-					drivesPerfInfo = append(drivesPerfInfo, madmin.DrivePerfInfo{
-						Path:  endpoint.Path,
-						Error: fmt.Sprintf("stat: %v", err),
-					})
-					continue
-				}
-				measurePath := pathJoin(minioMetaTmpBucket, mustGetUUID())
-				measure := func(path string) {
-					defer wg.Done()
-					driveInfo := madmin.DrivePerfInfo{
-						Path: path,
-					}
-					latency, throughput, err := disk.GetHealthInfo(ctx, path, pathJoin(path, measurePath))
-					if err != nil {
-						driveInfo.Error = fmt.Sprintf("health-info: %v", err)
-					} else {
-						driveInfo.Latency = latency
-						driveInfo.Throughput = throughput
-					}
-					drivesPerfInfo = append(drivesPerfInfo, driveInfo)
-				}
-				wg.Add(1)
+	for _, pool := range pools {
+		for _, endpoint := range pool.Endpoints {
+			if !endpoint.IsLocal {
+				continue
+			}
 
-				if parallel {
-					go measure(endpoint.Path)
+			if _, err := os.Stat(endpoint.Path); err != nil {
+				info = append(info, madmin.DiskPerfInfo{
+					Path:  endpoint.Path,
+					Error: err.Error(),
+				})
+				continue
+			}
+
+			workDir := pathJoin(endpoint.Path, minioMetaTmpBucket, mustGetUUID())
+			getHealthInfo := func(workDir string) {
+				latency, throughput, err := disk.GetHealthInfo(ctx, endpoint.Path, workDir)
+				if err != nil {
+					info = append(info, madmin.DiskPerfInfo{
+						Path:  workDir,
+						Error: err.Error(),
+					})
 				} else {
-					measure(endpoint.Path)
+					info = append(info, madmin.DiskPerfInfo{
+						Path: workDir,
+						Latency: madmin.Latency{
+							Avg:          round(latency.Avg, 3),
+							Max:          round(latency.Max, 3),
+							Min:          round(latency.Min, 3),
+							Percentile50: round(latency.Percentile50, 3),
+							Percentile90: round(latency.Percentile90, 3),
+							Percentile99: round(latency.Percentile99, 3),
+						},
+						Throughput: madmin.Throughput{
+							Avg:          uint64(round(throughput.Avg, 0)),
+							Max:          uint64(round(throughput.Max, 0)),
+							Min:          uint64(round(throughput.Min, 0)),
+							Percentile50: uint64(round(throughput.Percentile50, 0)),
+							Percentile90: uint64(round(throughput.Percentile90, 0)),
+							Percentile99: uint64(round(throughput.Percentile99, 0)),
+						},
```

Comment:
No. Throughput value is in bytes per second. Factional value to this is quite useless.

Reply:
I don't get why we need to create duplicate structures just to remove the fractional part from the values.

Reply:
This is exactly I was told to do i.e. do not modify existing structure in other places.

Reply:
I'm not asking to modify any existing structure. Just not sure why need a duplicate structure to have rounded values and lose the precision. I understand that the fractional part may not be too useful for decision making, but still not sure if we want to have new structures and do extra processing just to round off the numbers.

@harshavardhana @wlan0 your opinion please.

Reply:
@balamurugana @anjalshireesh perhaps the coding is making sure to avoid modifying the lower level calls - we still keep precision at a lower level but not worry about it when capturing health info. 

This looks like an okay trade-off.